qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1413,6 +1413,18 @@ def set_html(self, html, base_url=QUrl()):
         # percent encoded content is 2 megabytes minus 30 bytes.
         self._widget.setHtml(html, base_url)
 
+    def set_lifecycle_state(
+            self,
+            new_state: QWebEnginePage.LifecycleState
+    ) -> None:
+        """Set the lifecycle state of the current tab."""
+        if sip.isdeleted(self._widget):
+            log.webview.debug("Ignoring page lifecycle update for deleted widget")
+            return
+
+        log.webview.debug("Setting page lifecycle state of {} to {}".format(self.url().toString(), new_state))
```

Comment:
For the first parameter here, AbstractTab [already provdes](https://github.com/qutebrowser/qutebrowser/blob/ff3a4547e8e6e91f007eb0afd2dfc45dd8177e53/qutebrowser/browser/browsertab.py#L1361) a reasonable string representation of a tab we can use.

Reply:
How would I use that? Do I just pass `self` directly and it stringifies itself? :)

Reply:
Exactly! And while you're at it, could you rewrite this to use [f-strings](https://fstring.help/) instead?